### PR TITLE
Sparkles rework/publish

### PR DESCRIPTION
**Pull request by bogedal**
Will this work?

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,23 +1,38 @@
-name: Kirby release CI
+name: Publish to npm
 
 on:
-  push:
-    tags:
-      - '^v[0-9]+\.[0-9]+'
+  pull_request:
+    types: closed
```
Comment (Sobuno) [inline .github/workflows/publish.yml:5]: What if a PR does not target master but some other branch?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
         with:
           node-version: '10.16'
-      - name: Create npm config
-        run: echo '//registry.npmjs.org/:_authToken=${{secrets.NPM_PUBLISH_TOKEN}}' > ~/.npmrc
+      - name: Install npm
+        run: npm install
+      - name: Configure git
+        run: |
+          git config user.name "Kirby Bot"
+          git config user.email "<>"
```
Comment (Sobuno) [inline .github/workflows/publish.yml:28]: Is this legal?

**Review by Sobuno — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 2 above)
Comment (bogedal) [reply .github/workflows/publish.yml:28]: Will this work ... Yes
So in what sense do you mean legal?

File: .github/workflows/publish.yml (hunk 1 above)
Comment (bogedal) [reply .github/workflows/publish.yml:5]: May be an issue - I'll look into it

Comment (jakobe) [reply .github/workflows/publish.yml:5]: Good point @Sobuno 👍 🤔

Comment (bogedal) [reply .github/workflows/publish.yml:5]: Fixed? :)

Comment (jakobe) [inline .github/workflows/publish.yml:5]:
```suggestion
  pull_request:
    types: closed
    branches: master
```

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -1,23 +1,38 @@
...
+        with:
+          ref: refs/heads/master      
       - name: Install Node.js
         uses: actions/setup-node@v2
         with:
           node-version: '10.16'
-      - name: Create npm config
-        run: echo '//registry.npmjs.org/:_authToken=${{secrets.NPM_PUBLISH_TOKEN}}' > ~/.npmrc
+      - name: Install npm
+        run: npm install
```
Comment (jakobe) [inline .github/workflows/publish.yml:25]: Align with build workflow;
```suggestion
        run: npm ci
```

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -1,23 +1,38 @@
...
+      - name: Configure git
+        run: |
+          git config user.name "Kirby Bot"
+          git config user.email "<>"
+      - name: Create .npmrc
```
Comment (jakobe) [inline .github/workflows/publish.yml:30]: Consider naming:
```suggestion
      - name: Create npm config
```

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -1,23 +1,38 @@
...
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
+        run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
```
Comment (jakobe) [inline .github/workflows/publish.yml:33]: Simplify (revert/keep existing script)
```suggestion
        run: echo '//registry.npmjs.org/:_authToken=${{secrets.NPM_PUBLISH_TOKEN}}' > ~/.npmrc
```

File: .github/workflows/publish.yml [hunk 6]
```diff
@@ -1,23 +1,38 @@
...
+      - name: Bump version
+        run: npm version patch -m ":bookmark:Bumping version to %s"
       - name: Publish to npm
         run: sh ./scripts/publish.js
+      - name: Update github with version tag
```
Comment (jakobe) [inline .github/workflows/publish.yml:38]: Consider naming:
```suggestion
      - name: Push version bump to Github
```

File: .github/workflows/publish.yml [hunk 7]
```diff
@@ -1,23 +1,38 @@
...
+        run: git push HEAD:master --tags
```
Comment (jakobe) [inline .github/workflows/publish.yml:39]: Which is better? The existing "postversion" script is:
`"postversion": "git push && git push --tags",`
🤔

**Review by jakobe — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 7 above)
Comment (bogedal) [reply .github/workflows/publish.yml:37]: Lavet et 3. forslag :)

**Review by jakobe — APPROVED** [review]

**Review by jakobe — APPROVED** [review]

**Pull request merged**